politeiawww/mail/client_test.go

```diff
@@ -0,0 +1,297 @@
+		t.Errorf("warning emails list length want 1, got %v: %v",
+			len(fr.valid), fr.warning)
+	}
+
+	// Verify returned email history for noHistory user
+	eh, ok := fr.histories[userNoHistory]
+	switch {
+	case !ok:
+		t.Errorf("user with no email history was not found in the " +
+			"histories list")
+
+	case len(eh.Timestamps) != 1:
+		t.Errorf("histories length for user with no email history: "+
+			"want 1, got %v", len(eh.Timestamps))
+
+	case eh.LimitWarningSent:
+		t.Errorf("limit warning sent for user with no email history: " +
+			"want false, got true")
+	}
+
+	// Verify returned email history for underLimit user
+	eh, ok = fr.histories[userUnderLimit]
+	switch {
+	case !ok:
+		t.Errorf("user with email history under the rate limit was " +
+			"not found in the histories list")
+
+	case len(eh.Timestamps) != 2:
+		t.Errorf("histories length for user under the rate limit: "+
+			"want 2, got %v", len(eh.Timestamps))
+
+	case eh.LimitWarningSent:
+		t.Errorf("limit warning sent for user with email history " +
+			"under the rate limit: want false, got true")
+	}
+
+	// Verify returned email history for nearLimit user
+	eh, ok = fr.histories[userNearLimit]
+	switch {
+	case !ok:
+		t.Errorf("user with email history under the rate limit " +
+			"by one was not found in the histories list")
+
+	case len(eh.Timestamps) != 3:
+		t.Errorf("histories length for user under the rate limit "+
+			"by one: want 3, got %v", len(eh.Timestamps))
+
+	case !eh.LimitWarningSent:
+		t.Errorf("limit warning sent for user with email history " +
+			"under the rate limit by one: want true, got false")
+	}
+
+	// Verify returned email history for atLimitExpired user
+	eh, ok = fr.histories[userAtLimitExpired]
+	switch {
+	case !ok:
+		t.Errorf("user with email history at the rate limit but " +
+			"expired was not found in the histories list")
+
+	case len(eh.Timestamps) != 1:
+		t.Errorf("histories length for user under the rate limit: "+
+			"want 1, got %v", len(eh.Timestamps))
+
+	case eh.LimitWarningSent:
+		t.Errorf("limit warning sent for user with email history " +
+			"at the rate limit but expired: want false, got true")
+	}
+
+	// Verify the filtered histories does not contain unexpected
+	// histories.
+	if len(fr.histories) != 4 {
+		t.Errorf("filtered histories length: want 4, got %v",
+			len(fr.histories))
+	}
+}
+
+func TestFilterTimestamps(t *testing.T) {
+	// Setup test cases
+	var tests = []struct {
+		name    string
+		in      []int64
+		wantOut []int64
+	}{
+		{
+			"remove all timestamps",
+			[]int64{
+				time.Now().Add(-(24 * time.Hour)).Unix(),
+				time.Now().Add(-(36 * time.Hour)).Unix(),
+				time.Now().Add(-(42 * time.Hour)).Unix(),
+			},
+			[]int64{},
+		},
+		{
+			"remove stale timestamps",
+			[]int64{
+				time.Now().Add(-(12 * time.Hour)).Unix(),
+				time.Now().Add(-(26 * time.Hour)).Unix(),
+				time.Now().Add(-(28 * time.Hour)).Unix(),
+			},
+			[]int64{
+				time.Now().Add(-(12 * time.Hour)).Unix(),
```

Comment:
Its possible for the Unix timestamp to change in between the input and output creations, causing a test failure even when there shouldn't be one.  We don't want that.

Reply:
Fixed, testing for function output length now to avoid this case